libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/accounts/MapboxNavigationAccounts.kt

```diff
@@ -0,0 +1,43 @@
+package com.mapbox.services.android.navigation.v5.internal.accounts
+
+import android.content.Context
+import android.text.format.DateUtils
+import com.mapbox.services.android.navigation.v5.navigation.NavigationConstants.MAPBOX_SHARED_PREFERENCES
+
+class MapboxNavigationAccounts private constructor() {
```

Comment:
looks like this class simply wraps the skuGenerator calls and serves as a static factory for setup, should this class be an `object` that exposes a property `skuGenerator` with default setup? 


```suggestion
object MapboxNavigationAccounts {
 private const val MAU_TIMER_EXPIRE_THRESHOLD = 1
 private const val TRIPS_TIMER_EXPIRE_THRESHOLD = 2
 private const val TRIPS_REQUEST_COUNT_THRESHOLD = 5
 private const val TIMER_EXPIRE_AFTER = DateUtils.HOUR_IN_MILLIS / 1000
 lateinit var skuGenerator: SKUGenerator
@JvmStatic
fun init(context: Context) {
   val preferences = context.getSharedPreferences(MAPBOX_SHARED_PREFERENCES, Context.MODE_PRIVATE)
   this.skuGenerator = when (Billing.getInstance(context).getBillingType()) {
                Billing.BillingModel.MAU -> MauSku(preferences, TIMER_EXPIRE_AFTER * MAU_TIMER_EXPIRE_THRESHOLD)
                Billing.BillingModel.TRIPS -> TripsSku(preferences, TIMER_EXPIRE_AFTER * TRIPS_TIMER_EXPIRE_THRESHOLD, TRIPS_REQUEST_COUNT_THRESHOLD)
            }
  }
}
```
then `init` can be called at `ContentProvider#onCreate()` and all of your classes would have a singleton access to the sku generator.

Reply:
This class is already a `Singleton` as it has a private constructor and a `getInstance()` method. The reason `object` was not used is because they cannot have a constructor. We are on the same page.